src/app/modules/details/test-exec-details.component.ts

```diff
@@ -58,6 +61,30 @@ export class TestExecDetailsComponent implements OnInit, OnDestroy {
     }
   }
 
+  get logLevel(): SelectableLogLevel {
+    return this.logLevel_;
+  }
+
+  set logLevel(logLevel: SelectableLogLevel) {
+    if (this.logLevel_ !== logLevel) {
+      const previousLogLevel = this.logLevel_;
+      this.logLevel_ = logLevel;
+      this.retrieveLog(previousLogLevel);
+    }
+  }
+
+  private async retrieveLog(previousLogLevel: SelectableLogLevel) {
```

Comment:
I'd drop previous level from parameter list. if it fails to retrieve any detail, this is hardly a log level problem, is it?

Reply:
Right, my intention was to reset the button to the previous log level setting in case of a problem while requesting the log on a different level. But that should happen if an exception occurs, not if a positive but empty response is received.

Anyways, I'll remove the reset to the previous level. I will add a try-catch, though, but only to log an error into the console (makes no difference to the user… do we want a user-facing message in case of a problem?)